libraries/ms-common/src/stm32f0xx/crc32.c

```diff
@@ -0,0 +1,26 @@
+#include "crc32.h"
+#include "log.h"
+#include "stm32f0xx.h"
+
+StatusCode crc32_init(void) {
+  RCC_AHBPeriphClockCmd(RCC_AHBPeriph_CRC, ENABLE);
+  CRC_ReverseInputDataSelect(CRC_ReverseInputData_32bits);
+  CRC_ReverseOutputDataCmd(ENABLE);
+
+  return STATUS_CODE_OK;
+}
+
+uint32_t crc32_arr(const uint8_t *buffer, size_t buffer_len) {
+  CRC_ResetDR();
+
+  size_t num_u32 = buffer_len / sizeof(uint32_t);
+  size_t remaining_bytes = buffer_len % sizeof(uint32_t);
+  uint32_t crc = CRC_CalcBlockCRC((const uint32_t *)buffer, num_u32);
```

Comment:
Maybe add comments? I had to go look at the underlying functions because this interface is really weird coming from C++ versions. Even something like "consume buffer of uint32" then "consume remaining bytes" for the separate subsections.

Reply:
Sure, it's because the underlying peripheral consumes words by default. The only reason this even works is because the 072 series has the ability to use non-words. :)